### PR TITLE
[listener] Break out of while loop if accept was a success.

### PR DESCRIPTION
**Pull request by NathanFrench**
We must break out of this while loop if the accept() did not return an error in order to avoid potential stalls.

### CODE REVIEW

**Comment by azat** [comment]
Can you be more verbose? For example: when accept() returns which error we should exit from the accept loop? We already will exit from from it the error is not retriable.

File: listener.c [hunk 1]
```diff
@@ -429,6 +429,11 @@ listener_read_cb(evutil_socket_t fd, short what, void *p)
 			UNLOCK(lev);
 			return;
 		}
+
+        if (new_fd != -1) {
+           UNLOCK(lev);
+           return;
+        }
```
Comment (azat) [inline listener.c]: `s/space/tab/`

**Comment by azat** [comment]
@NathanFrench Can you be more verbose? I.e. what are the reason for this?

(even if this should be added, it requires extra flag to me)

**Comment by NathanFrench** [comment]
I'm going to close this in lieu of using a flag to optionally do this. This can lead to problems where the system is under a high rate of new connections, but still able to accept before the accept queue gets too high. Essentially 50% of `accept()` will error on the second loop with tcp.

**Pull request closed without merging**